### PR TITLE
Add sso

### LINKED ISSUE #7: Website seems public, but videos/thumbnails are not
https://masterclasses.canonical.com/ seems to be a public website now, but the videos and thumbnails are not.

I'm not sure if it's meant to be public, so maybe it should be behind SSO?

### PR DESCRIPTION
**Pull request by andesol**
Add SSO

## QA

- Run the project locally with `dotrun`
- Go to http://0.0.0.0:9000/ (or [demo](https://masterclasses-canonical-com-8.demos.haus/)) and check that you're redirected to SSO

## Issue

Fixes #7

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,4 +1,5 @@
 PORT=9000
 FLASK_DEBUG=true
 SECRET_KEY=secret_key
-DEVEL=True
\ No newline at end of file
+DEVEL=True
+OPENID_LAUNCHPAD_TEAM=canonical-content-people
```
[the remark this answers is not in the document]
Comment (andesol) [reply .env:5]: @bartaz Interesting. I have to say that I never got to properly understand how these groups work. This is the one we're using for hiring.c.c. and specs.c.c, so I expected similar users here. If I try to log with the default group ("canonical") it fails.

**Comment by andesol** [comment]
Closing this one as this website is meant to be public per #7

File: .env (hunk 1 above)
[the remark this answers is not in the document]
Comment (bartaz) [reply .env:5]: Oh, OK. If that's what we use for specs, I guess it's correct.

**Comment by aaryanporwal** [comment]
Hey @andesol reopening this PR.
@anthonydillon and I have decided to put the website behind SSO for now.

**Comment by anthonydillon** [comment]
Do any of the sites use the canonical group? I know it's difficult as it's a private group but it's well maintained.

File: .env (hunk 1 above)
[the remark this answers is not in the document]
Comment (andesol) [reply .env:5]: Just to clarify, this is for local development. In production SSO should request the `canonical` group

**Comment by edlerd** [comment]
> Do any of the sites use the canonical group? I know it's difficult as it's a private group but it's well maintained.

Specs.c.c is [using the canonical group](https://github.com/canonical/specs.canonical.com/blob/main/webapp/sso.py#L8C23-L8C44).

**Review by edlerd — APPROVED** [review]
LGTM, when testing. Needs a rebase for the demo to work. When I rebased locally it worked fine.

**Pull request merged**